plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/IcebergPositionDeletePageSink.java

```diff
@@ -0,0 +1,189 @@
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.plugin.hive.util.ConfigurationUtils.toJobConf;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.apache.iceberg.io.DeleteSchemaUtil.pathPosSchema;
+
+public class IcebergPositionDeletePageSink
+        implements ConnectorPageSink
+{
+    private static final int MAX_PAGE_POSITIONS = 4096;
+    private static final Schema POSITION_DELETE_SCHEMA = pathPosSchema();
+
+    private final String dataFilePath;
+    private final PartitionSpec partitionSpec;
+    private final Optional<PartitionData> partition;
+    private final String outputPath;
+    private final JsonCodec<CommitTaskData> jsonCodec;
+    private final IcebergFileWriter writer;
+    private final IcebergFileFormat fileFormat;
+
+    private long writtenBytes;
+    private long memoryUsage;
+    private long validationCpuNanos;
+    private boolean writtenData;
+
+    public IcebergPositionDeletePageSink(
+            String dataFilePath,
+            PartitionSpec partitionSpec,
+            Optional<PartitionData> partition,
+            LocationProvider locationProvider,
+            IcebergFileWriterFactory fileWriterFactory,
+            HdfsEnvironment hdfsEnvironment,
+            HdfsContext hdfsContext,
+            JsonCodec<CommitTaskData> jsonCodec,
+            ConnectorSession session,
+            IcebergFileFormat fileFormat)
+    {
+        this.dataFilePath = requireNonNull(dataFilePath, "dataFilePath is null");
+        this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
+        this.partitionSpec = requireNonNull(partitionSpec, "partitionSpec is null");
+        this.partition = requireNonNull(partition, "partition is null");
+        this.fileFormat = requireNonNull(fileFormat, "file is null");
+        // prepend query id to a file name so we can determine which files were written by which query. This is needed for opportunistic cleanup of extra files
+        // which may be present for successfully completing query in presence of failure recovery mechanisms.
+        String fileName = fileFormat.toIceberg().addExtension(session.getQueryId() + "-" + randomUUID());
+        this.outputPath = partition
+                .map(partitionData -> locationProvider.newDataLocation(partitionSpec, partitionData, fileName))
+                .orElseGet(() -> locationProvider.newDataLocation(fileName));
+        JobConf jobConf = toJobConf(hdfsEnvironment.getConfiguration(hdfsContext, new Path(outputPath)));
+        this.writer = fileWriterFactory.createPositionDeleteWriter(new Path(outputPath), POSITION_DELETE_SCHEMA, jobConf, session, hdfsContext, fileFormat.toIceberg());
```

Comment:
nit: `createPositionDeleteWriter` could take `fileFormat` and do `toIceberg()` on its own (or not at all).
-- This is Trino code, not Trino/Iceberg boundary